### PR TITLE
Adding features:

### PR DESCRIPTION
**Pull request by jojobote**
Merged PR case
Closed PR case
Branch name checking

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,41 +1,94 @@
+# ssh deployments
 
-This GitHub Action deploys your current issue branch to a remote Pantheon repo, using NodeJS.
+Deploy code with rsync over ssh, using NodeJS.
```
Comment (clobee) [inline README.md]: You are bringing the doc back to before my update :cry:

File: dist/index.js [hunk 2]
```diff
@@ -9662,13 +9675,42 @@ const pantheonDeploy = (() => {
...
         } catch (error) {
             core.setFailed(error.message);
             process.abort();
         }
     };
 
+    async function merge(machineToken, pantheonRepoName, pullRequest) {
+        try {
+
+            await exec.exec('curl -O https://raw.githubusercontent.com/pantheon-systems/terminus-installer/master/builds/installer.phar');
```
Comment (clobee) [inline dist/index.js:9733]: This is done quite few times in your script. Can be done once

File: dist/index.js [hunk 3]
```diff
@@ -9662,13 +9675,42 @@ const pantheonDeploy = (() => {
...
+        }
+    }
+
+    async function close(machineToken, pantheonRepoName, pullRequest) {
+        try {
+
+            await exec.exec('curl -O https://raw.githubusercontent.com/pantheon-systems/terminus-installer/master/builds/installer.phar');
+            await exec.exec('sudo php installer.phar install'); // Sudo is required in order to install bin/terminus.
+            await exec.exec('terminus', ['auth:login', `--machine-token=${ machineToken }`]);
+            await exec.exec('terminus', ['multidev:delete', pantheonRepoName, pullRequest.head.ref]);
```
Comment (clobee) [inline dist/index.js:9760]: Only this action seems required in this close() function the rest is repeated

File: dist/index.js [hunk 4]
```diff
@@ -9677,17 +9719,20 @@ const pantheonDeploy = (() => {
             await exec.exec('terminus', ['auth:login', `--machine-token=${ machineToken }`]);
             await exec.exec('terminus', ['multidev:create', pantheonRepoName, pullRequest.head.ref]);
 
-            output = JSON.stringify(child_process.execSync(`terminus env:view --print ${ pantheonRepoName }.${ pullRequest.head.ref }`));
+            const output = JSON.stringify(child_process.execSync(`terminus env:view --print ${ pantheonRepoName }.${ pullRequest.head.ref }`));
+            console.log('URL to access the multidev is : ' . output);
```
Comment (clobee) [inline dist/index.js]: "\n " in the console log

File: src/index.js [hunk 5]
```diff
@@ -16,16 +16,29 @@ console.log('GITHUB_WORKSPACE', GITHUB_WORKSPACE);
...
+        checkBranch(pullRequest.head.ref, strictBranchName);
         gitBranch(pantheonRepoURL, pullRequest);
         buildMultiDev(machineToken, pantheonRepoName, pullRequest);
     };
 
+    const checkBranch = (prName, strictBranchName) => {
+        if (prName.length > 11) {
+            core.setFailed("Branch name is too long to create a multidev. Branch names need to be 11 characters or less.");
+            process.abort();
+        } else if (strictBranchName == "strict" && prName.match(/[A-z]*-[0-9]*/)) {
```
Comment (clobee) [inline src/index.js]: How do you manage 2 dudes working on the same tickets... So they shoudl be able to use same branch with a differentiator Eg: CHIV-123 & CHIV-123-2

File: dist/index.js (hunk 4 above)
Comment (jojobote) [reply dist/index.js]: Yeah they're everywhere.. I think they are needed if we want them to be properly displayed ?

File: action.yml [hunk 6]
```diff
@@ -1,7 +1,10 @@
 name: "Pantheon deploy"
-description: "This GitHub Action deploys your current issue branch to a remote Pantheon repo."
+description: "NodeJS action to deploy a branch to Pantheonfor FAST deployment with rsync/ssh"
```
Comment (clobee) [inline action.yml]: Issue with the description

File: action.yml [hunk 7]
```diff
@@ -11,6 +14,9 @@ inputs:
   REMOTE_REPO_NAME:
     description: "Remote GIT Repo Name"
     required: true
+  SSH_AUTH_SOCK:
+    description: 'Where to place the SSH Agent auth socket'
+    default: /tmp/ssh-auth.sock
```
Comment (clobee) [inline action.yml]: This should not be here

File: action.yml [hunk 8]
```diff
@@ -1,7 +1,10 @@
 name: "Pantheon deploy"
-description: "This GitHub Action deploys your current issue branch to a remote Pantheon repo."
+description: "NodeJS action to deploy a branch to Pantheonfor FAST deployment with rsync/ssh"
 author: "irishdistillers"
 inputs:
```
Comment (clobee) [inline action.yml]: Missing other inputs E.g: STRICT_BRANCH_NAMES, PR_STATE ...

File: dist/index.js [hunk 9]
```diff
@@ -9637,13 +9637,80 @@ console.log('GITHUB_WORKSPACE', GITHUB_WORKSPACE);
 const pantheonDeploy = (() => {
 
     const init = ({
+        prState,
```
Comment (clobee) [inline dist/index.js:9640]: You are using prState here but pullRequest further down. Could we maybe use pullRequestState?

File: action.yml (hunk 8 above)
Comment (jojobote) [reply action.yml]: What is going on with my local branch resetting files even though I update them all the time ... Fixing

**Review by clobee — COMMENTED** [review]
Looks good 👍  few little comments

**Review by clobee — APPROVED** [review]

**Pull request merged**